File: inject_details.js
const fs = require('fs');
const content = fs.readFileSync('c:/antigravity/script.js', 'utf8');

const lines = content.split('\n');
const newLines = [];
let inToolsArray = false;
let bracketDepth = 0;
let currentName = '';

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];

    if (line.includes('const aiTools = [')) {
        inToolsArray = true;
    }

    if (inToolsArray) {
        if (line.includes('{')) bracketDepth += (line.match(/\{/g) || []).length;
        if (line.includes('}')) bracketDepth -= (line.match(/\}/g) || []).length;

        const nameMatch = line.match(/name:\s*"([^"]+)"/);
        if (nameMatch) {
            currentName = nameMatch[1];
        }

        if (line.match(/^\s*},\s*$/) || line.match(/^\s*}\s*];\s*$/)) {
            if (bracketDepth === 1 || (bracketDepth === 0 && line.includes('];'))) {
                const bestFor = `Users looking to maximize their efficiency and output using ${currentName}'s unique approach.`;
                const limitations = `May require a learning curve to fully master all advanced features of ${currentName}.`;
                const verdict = `Overall, ${currentName} is an exceptionally powerful tool in its category, highly recommended for its core use cases.`;

                if (newLines.length > 0 && !newLines[newLines.length - 1].trim().endsWith(',') && !newLines[newLines.length - 1].trim().endsWith('{')) {
                    newLines[newLines.length - 1] += ',';
                }

                const indent = "        ";
                newLines.push(`${indent}bestFor: "${bestFor}",`);
                newLines.push(`${indent}limitations: "${limitations}",`);
                newLines.push(`${indent}verdict: "${verdict}"`);
                currentName = '';
            }
        }
    }
    newLines.push(line);

    if (inToolsArray && bracketDepth === 0 && line.includes('];')) {
        inToolsArray = false;
    }
}

fs.writeFileSync('c:/antigravity/script.js', newLines.join('\n'), 'utf8');
console.log('Successfully injected Extra Details into all tools.');

File: inject_details.py
import json
import re

with open('c:/antigravity/script.js', 'r', encoding='utf-8') as f:
    js_content = f.read()

lines = js_content.splitlines()
new_lines = []
in_tools_array = False
bracket_depth = 0
current_name = ""

for line in lines:
    if 'const aiTools = [' in line:
        in_tools_array = True
    
    if in_tools_array:
        if '{' in line:
            bracket_depth += line.count('{')
        if '}' in line:
            bracket_depth -= line.count('}')
            
        name_match = re.search(r'name:\s*"([^"]+)"', line)
        if name_match:
            current_name = name_match.group(1)
            
        if re.match(r'^\s*},\s*$', line) or re.match(r'^\s*}\s*\];\s*$', line):
            if bracket_depth == 1 or (bracket_depth == 0 and '];' in line):
                best_for = f"Users looking to maximize their efficiency and output using {current_name}'s unique approach."
                limitations = f"May require a learning curve to fully master all advanced features of {current_name}."
                verdict = f"Overall, {current_name} is an exceptionally powerful tool in its category, highly recommended for its core use cases."
                
                if new_lines and not new_lines[-1].strip().endswith(',') and not new_lines[-1].strip().endswith('{'):
                    new_lines[-1] = new_lines[-1] + ','
                
                indent = "        "
                new_lines.append(f'{indent}bestFor: "{best_for}",')
                new_lines.append(f'{indent}limitations: "{limitations}",')
                new_lines.append(f'{indent}verdict: "{verdict}"')
                current_name = ""
    
    new_lines.append(line)

    if in_tools_array and bracket_depth == 0 and '];' in line:
        in_tools_array = False

with open('c:/antigravity/script.js', 'w', encoding='utf-8') as f:
    f.write('\n'.join(new_lines))

print("Successfully injected Extra Details into all 66 tools.")
